apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -69,12 +69,16 @@ impl Oracle {
     /// ledger. Returns a boolean indicating that all sent
     /// successfully. If false is returned, the receiver
     /// has hung up.
-    fn send(&self, events: Vec<EthereumEvent>) -> bool {
-        events
-            .into_iter()
-            .map(|event| self.sender.send(event))
-            .all(|res| res.is_ok())
-            && !self.sender.is_closed()
+    ///
+    /// N.B. this will block if the internal channel buffer
+    /// is full.
+    async fn send(&self, events: Vec<EthereumEvent>) -> bool {
+        for event in events.into_iter() {
+            if self.sender.send(event).await.is_err() {
+                return false;
+            }
```

Comment:
```suggestion
            match self.sender.send(event).await {
                Ok(()) => continue,
                Err(error) => {
                    tracing::warn!(
                        ?error,
                        "Failed to send Ethereum event to ledger",
                    );
                    return false;
                }
            }
```

If it fails, let's log the error. It also looks like we will actually shut down the ledger if this fn returns false?

https://github.com/anoma/namada/blob/64f73c7186d78d73f80f51a5851ff24468d9ad9b/apps/src/lib/node/ledger/ethereum_node/oracle.rs#L199-L205

but we can fix this in another PR

Reply:
We probably want to make an issue to refactor how this works a bit separately. We could return a `Result` or smth indicating how many of the `Vec<EthereumEvent>` we were able to send, and then the caller should wait and try again in a bit (when the channel might have some free space)

Reply:
Since we have to shut down here, I don't think the error message is helpful since a better error message will be immediately issued at line 199. I do think I can remove the `!self.sender.is_closed()` as that is now redundant.

Reply:
ahh right, so `.send()` will hang until there's space (compared to `try_send()` which errors if there isn't space), that makes sense. The error returned by `.send` could still be useful as it indicates the `EthereumEvent` that failed to send (so we should be able to work out how many of `Vec<EthereumEvent>`s we managed to actually send). Really this fn should return a `Result` though instead of `bool` and then the error could be logged by the caller with extra context, but we could refactor this later